src/comm/TcpAcceptor.cc

```diff
@@ -354,13 +357,9 @@ Comm::TcpAcceptor::oldAccept(Comm::ConnectionPointer &details)
 
         if (ignoreErrno(errcode) || errcode == ECONNABORTED) {
             debugs(50, 5, status() << ": " << xstrerr(errcode));
-            return Comm::NOMESSAGE;
-        } else if (errcode == ENFILE || errcode == EMFILE) {
-            debugs(50, 3, status() << ": " << xstrerr(errcode));
-            return Comm::COMM_ERROR;
+            return false;
         } else {
-            debugs(50, DBG_IMPORTANT, "ERROR: failed to accept an incoming connection: " << xstrerr(errcode));
-            return Comm::COMM_ERROR;
+            throw TextException(ToSBuf("Failed to accept an incoming connection: ", xstrerr(errcode)), Here());
```

Comment:
> > PR description: Silent lack of accept() after an ENFILE or EMFILE failure.
 
> Amos: So-called "silence" for these errors is not a problem.

It is a fairly serious problem IMO: We just spent several man/hours and almost a week troubleshooting a problematic Squid exactly because those failures did not result in any messages in cache.log. It should have taken us about 10 minutes, but without those cache.log messages, it takes a lot more time to figure out what is actually going on.

> Lack of accept() is the actual problem.

Yes, for some of these errors it is indeed, but the two problems (i.e. silence and lack of accept()) are not mutually exclusive. Both problems are real and co-exist.


> > Silent at level-0 lack of accept() after logging a level-1 ERROR. This
> > problem was specific to ALL,0 and similar (rare) configurations.
 
> This is correctly obeying admin preferences. There is no problem here.

I disagree: First of all, when Squid stops accepting connections on a port, it is a level-0 (i.e. _critical_) error. Some would even argue that such  an error should be fatal (but let's not have that debate here and now). The lack of the corresponding level-0 error in the official code is a bug. Secondly, the level-1 ERROR which _is_ logged in some cases, is _not_ about stopping to accept() future connections (i.e. it is _not_ the missing message that the third bullet point talks about). That level-1 ERROR looks like a regular transaction error to most admins.

> Also, please define what you mean by "Silent". The dictionary meaning of that word contradicts the code - all the above cases have cache.log and/or access.log entries associated with the relevant event.

I think debugging modes (i.e. level-2+) should not count by default -- they are not for admins -- but I made that explicit by adding "non-debugging" to the corresponding paragraph above the bullet points. With that out of the way:

The first two of the three no-accept cases do not have the corresponding cache.log messages at all. The third no-accept case does not have a level-0 message (as detailed in the PR description). The paragraph before the three bullet points already summarizes them as "no error messages in cache.log", effectively confirming what "silent" means in this context. If that is not enough, a [dictionary](https://www.merriam-webster.com/dictionary/silent) meaning of that word -- "making no mention" -- matches the official code AFAICT.

I do not think we should account for access.log entries in this context because a lack of an accept is a Squid port-wide problem rather than a transaction-specific problem. Admins should not be expected to fish for reasons why Squid may have stopped performing one of its primary functions in an access.log of a busy proxy, especially when we do not detail those access.log records enough (yet) to know whether a particular accept(2) error resulted in the lack of further accept(2) calls.

@yadij, I plan to merge this PR this weekend with the current (already adjusted with "non-debugging") PR description unless you speed that up or stop me. If you stop me, please suggest the wording you want to see (or simply edit the current one).